Source/WebCore/html/CustomPaintCanvas.cpp

```diff
@@ -77,27 +72,20 @@ void CustomPaintCanvas::replayDisplayList(GraphicsContext& target)
         return;
     auto& imageTarget = image->context();
     imageTarget.translate(-clipBounds.location());
-    replayDisplayListImpl(imageTarget);
+    if (m_context)
```

Comment:
No, that is a change of behavior and incorrect operation in case, for example, when drawing with certain compositing operations.

Reply:
If `m_context` is nullptr then no `DisplayList` will be replayed back to the `imageTarget`. This will make `image` a transparent empty `ImageBuffer`. Why do we still need to draw this transparent empty `ImageBuffer`? What compositing operations will have an effect on drawing a transparent empty `ImageBuffer`?

Any way I think an early return here is an optimization but this should not affect the correctness.

Reply:
If the context has Copy compositing mode instead of SourceOver, the skip would mean the drawing buffer would be left uninitialized instead of initialized with transparent.
As said, the previous code was like that, and I'd like to keep it similar. The CustomPaintCanvas is buggy in various ways related to here, so if it is ever needed, it needs to be re-implemented anyway.